### PR TITLE
Remove old properties

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes https://github.com/aspnet/AspNetCore/issues/5449. We have some old properties that aren't needed anymore. Lets kill them.

### CODE REVIEW

**Review by natemcmaster — APPROVED** [review]

File: src/Servers/IIS/IIS/samples/NativeIISSample/NativeIISSample.csproj [hunk 1]
```diff
@@ -5,7 +5,6 @@
   <PropertyGroup>
     <TargetFramework>netcoreapp3.0</TargetFramework>
     <InProcessTestSite>true</InProcessTestSite>
-    <AspNetCoreHostingModel>inprocess</AspNetCoreHostingModel>
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/samples/NativeIISSample/NativeIISSample.csproj]: @pranavkm if this is now the default then does the other sample need to be updated for out-of-proc?

Comment (pranavkm) [reply src/Servers/IIS/IIS/samples/NativeIISSample/NativeIISSample.csproj]: Did you mean @pakrym ?

Comment (pakrym) [reply src/Servers/IIS/IIS/samples/NativeIISSample/NativeIISSample.csproj]: Do we have any samples that explicitly rely on running out-of-proc?

Comment (ryanbrandenburg) [reply src/Servers/IIS/IIS/samples/NativeIISSample/NativeIISSample.csproj]: @Tratcher that shouldn't stop this PR right? It's just a followup item for @pakrym.

Comment (Tratcher) [reply src/Servers/IIS/IIS/samples/NativeIISSample/NativeIISSample.csproj]: This one: https://github.com/aspnet/AspNetCore/blob/b5f21b42be204f6c0a97a2f12bdfeef5d9be3759/src/Servers/IIS/IISIntegration/samples/IISSample/IISSample.csproj

**Comment by ryanbrandenburg** [comment]
I'm going to merge in ~1 hour if there are no objections.

**Pull request merged**